lib/ReactGridLayout.jsx

```diff
@@ -250,10 +250,18 @@ export default class ReactGridLayout extends React.Component {
 
     // Set state
     this.setState({
-      layout: compact(layout, this.props.verticalCompact),
       activeDrag: null,
       oldDragItem: null
     });
+
+    // Set State and callback, but only if layout changed
+    const newLayout = compact(layout, this.props.verticalCompact);
+    if (!isEqual(newLayout, this.state.layout)) {
```

Comment:
No worries - going to make some changes soon anyway, I'll put it in as part of that.